--- /etc/makepkg.conf	2020-07-01 09:52:38.000000000 +0800
+++ /home/sosiristseng/.makepkg.conf	2020-10-23 15:58:33.355354622 +0800
@@ -37,12 +37,12 @@

 #-- Compiler and Linker Flags
 CPPFLAGS="-D_FORTIFY_SOURCE=2"
-CFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fno-plt"
-CXXFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fno-plt"
+CFLAGS="-march=native -O2 -pipe -fno-plt"
+CXXFLAGS="${CFLAGS}"
 LDFLAGS="-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now"
-#RUSTFLAGS="-C opt-level=2"
+RUSTFLAGS="-C opt-level=2 -C target-cpu=native"
 #-- Make Flags: change this for DistCC/SMP systems
-#MAKEFLAGS="-j2"
+MAKEFLAGS="-j$(nproc)"
 #-- Debugging flags
 DEBUG_CFLAGS="-g -fvar-tracking-assignments"
 DEBUG_CXXFLAGS="-g -fvar-tracking-assignments"
@@ -83,7 +83,7 @@
 #-- libtool:    Leave libtool (.la) files in packages
 #-- staticlibs: Leave static library (.a) files in packages
 #-- emptydirs:  Leave empty directories in packages
-#-- zipman:     Compress manual (man and info) pages in MAN_DIRS with gzip
+#-- zipman:     Compress manual (man and info) pages in MAN_DIRS with pigz
 #-- purge:      Remove files specified by PURGE_TARGETS
 #-- debug:      Add debugging flags as specified in DEBUG_* variables
 #
@@ -129,10 +129,10 @@
 # COMPRESSION DEFAULTS
 #########################################################################
 #
-COMPRESSGZ=(gzip -c -f -n)
+COMPRESSGZ=(pigz -c -f -n)
 COMPRESSBZ2=(bzip2 -c -f)
-COMPRESSXZ=(xz -c -z -)
-COMPRESSZST=(zstd -c -z -q -)
+COMPRESSXZ=(xz -c -z - --threads=0)
+COMPRESSZST=(zstd -c -z -q - --threads=0)
 COMPRESSLRZ=(lrzip -q)
 COMPRESSLZO=(lzop -q)
 COMPRESSZ=(compress -c -f)
